File: src/models/grades-overview.ts
import type { Grade } from "./grade";
import type { GradeValue } from "./grade-value";
import type { SubjectAverages } from "./subject-averages";

export type GradesOverview = Readonly<{
  subjectsAverages: SubjectAverages[]
  overallAverage?: GradeValue
  classAverage?: GradeValue
  grades: Grade[]
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/GradesOverview.kt
package ink.literate.pawnote.models

data class GradesOverview(
    val subjectsAverages: List<SubjectAverages>,
    val overallAverage: GradeValue? = null,
    val classAverage: GradeValue? = null,
    val grades: List<Grade>
)
